app/models/host/base.rb

```diff
@@ -134,6 +134,8 @@ def import_facts(facts)
       # we are not importing facts for hosts in build state (e.g. waiting for a re-installation)
       raise ::Foreman::Exception.new('Host is pending for Build') if build?
 
+      facts[:domain].downcase!
```

Comment:
good catch, will add test for this which should prevent breaking in future as well